File: test/lit/zig/if_stmt.zig
// RUN: %cot emit-cir %s | %FileCheck %s

pub fn clamp(x: i32) i32 {
    var result: i32 = x;
    if (x > 100) {
        result = 100;
    }
    return result;
}

// CHECK-LABEL: func.func @clamp
// CHECK: cir.alloca i32 : !cir.ptr
// CHECK: cir.cmp sgt
// CHECK: cir.condbr
// CHECK: cir.store
// CHECK: cir.br
// CHECK: cir.load
// CHECK: return

File: test/lit/ts/if_stmt.ts
// RUN: %cot emit-cir %s | %FileCheck %s

// Phase 2: if/else statement

function abs(x: number): number {
    if (x > 0) {
        return x;
    } else {
        return -x;
    }
}

// CHECK-LABEL: func.func @abs
// CHECK: cir.cmp sgt
// CHECK: cir.condbr
// CHECK: return
// CHECK: cir.neg
// CHECK: return
